app/src/org/commcare/utils/ConnectivityStatus.java

```diff
@@ -23,4 +49,21 @@ public static boolean isNetworkAvailable(Context context) {
         NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
         return activeNetworkInfo != null && activeNetworkInfo.isConnectedOrConnecting();
     }
+
+    public static NetworkState checkCaptivePortal() throws IOException {
+        // CommCare has its own URL for detecting Captive Portals.
+        // It must return an HTTP status code of 200 and a body containing "success".
+        String captivePortalURL = "http://www.commcarehq.org/serverup.txt";
```

Comment:
`www.commcarehq.org` isn't the only CommCare server on the internet, so we should be wary of reporting things like "CommCare Down." 

It's fine for us to use an "owned" URL to detect a network condition (since we are reporting something about the user's local connection), but if we are going to report on Server Status it should always be in reference to a Real Server that the user has attempted to communicate with, not our primary prod one.

Reply:
yeah, good catch.